File: WEEK6/pset6/readability/readability.py
from cs50 import get_string

def display(index) :
    if index < 1 :
        print("Before Grade 1")
    elif index > 16 :
        print("Grade 16+")
    else :
        print(f"Grade {index}")

def Coleman_Liau_formula(text) :
    text1 = text.split()
    sentences = 0
    words = len(text1)
    letters = 0
    for word in text1 :
        if word[-1] in ["!", "?", "."] :
            sentences += 1
            letters += len(word[:-1])
        else :
            for c in word :
                if c.isalpha() :
                    letters += 1
    index = 0.0588 * (letters/(words/100)) - 0.296 * (sentences/(words/100)) - 15.8
    return round(index)

def main() :
    text = get_string("Text: ")
    index = Coleman_Liau_formula(text)
    display(index)

main()

File: WEEK2/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
int main(void) {

    string text = get_string("Text : ");
    float letter = 0, word = 0, sentence = 0;

    for(int i = 0, n = strlen(text); i < n; i++) {
        int ascii = (int) toupper(text[i]);
        if (ascii >= 65 && ascii <= 90) {
            letter += 1;
        }
        else if (ascii == 32) {
            word += 1;
        }
        else if (ascii == 46 || ascii == 33 || ascii == 63) {
            sentence += 1;
        }
    }
    word += 1;

    float index = 0.0588 * (letter/(word/100)) - 0.296 * (sentence/(word/100)) - 15.8;

    if (index < 1) {
        printf("Before Grade 1\n");
    }
    else if (index > 16) {
        printf("Grade 16+\n");
    }
    else {
        printf("Grade %d\n", (int) round(index));
    }
}

